src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -20,129 +19,145 @@
      */
     public StorageFile() throws InvalidStorageFilePathException {
-        this(DEFAULT_STORAGE_FILEPATH);
+	this(DEFAULT_STORAGE_FILEPATH);
     }
 
     /**
-     * @throws InvalidStorageFilePathException if the given file path is invalid
+     * @throws InvalidStorageFilePathException
+     *             if the given file path is invalid
      */
     public StorageFile(String filePath) throws InvalidStorageFilePathException {
-        try {
-            jaxbContext = JAXBContext.newInstance(AdaptedAddressBook.class);
-        } catch (JAXBException jaxbe) {
-            throw new RuntimeException("jaxb initialisation error");
-        }
-
-        path = Paths.get(filePath);
-        if (!isValidPath(path)) {
-            throw new InvalidStorageFilePathException("Storage file should end with '.txt'");
-        }
+	try {
+	    jaxbContext = JAXBContext.newInstance(AdaptedAddressBook.class);
+	} catch (JAXBException jaxbe) {
+	    throw new RuntimeException("jaxb initialisation error");
+	}
+
+	path = Paths.get(filePath);
+	if (!isValidPath(path)) {
+	    throw new InvalidStorageFilePathException("Storage file should end with '.txt'");
+	}
     }
 
     /**
-     * Returns true if the given path is acceptable as a storage file.
-     * The file path is considered acceptable if it ends with '.txt'
+     * Returns true if the given path is acceptable as a storage file. The file
+     * path is considered acceptable if it ends with '.txt'
      */
     private static boolean isValidPath(Path filePath) {
-        return filePath.toString().endsWith(".txt");
+	return filePath.toString().endsWith(".txt");
     }
 
     /**
      * Saves all data to this storage file.
      *
-     * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws StorageOperationException
+     *             if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
-        /* Note: Note the 'try with resource' statement below.
-         * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
-         */
-        try (final Writer fileWriter =
-                     new BufferedWriter(new FileWriter(path.toFile()))) {
-
-            final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
-            final Marshaller marshaller = jaxbContext.createMarshaller();
-            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-            marshaller.marshal(toSave, fileWriter);
-
-        } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
-        } catch (JAXBException jaxbe) {
-            throw new StorageOperationException("Error converting address book into storage format");
-        }
+    public boolean save(AddressBook addressBook) throws StorageOperationException {
+
+	/*
```

Comment:
The `FileDeletedException` should be thrown from here instead. You should check if the file exists here. If not found, throw the exception to let it bubble up to the caller of this method.